### PR TITLE
Adding some idp related info to dev settings

### PR DESCRIPTION
**Pull request by wmathurin**

### CODE REVIEW

File: libs/SalesforceSDK/src/com/salesforce/androidsdk/app/SalesforceSDKManager.java [hunk 1]
```diff
@@ -636,6 +639,28 @@ public boolean isIDPLoginFlowEnabled() {
...
     }
 
+    /**
+     * Returns whether this application is configured as a Identity Provider
+     * @return True - if
+     */
+
+    public boolean isIdentityProvider() {
+        try {
+            PackageInfo packageInfo = context.getPackageManager().getPackageInfo(context.getPackageName(), PackageManager.GET_ACTIVITIES);
```
Comment (bhariharan) [inline libs/SalesforceSDK/src/com/salesforce/androidsdk/app/SalesforceSDKManager.java]: Does this require any special perms? Does it work on a `RELEASE` build as well?

File: libs/SalesforceSDK/src/com/salesforce/androidsdk/app/SalesforceSDKManager.java [hunk 2]
```diff
@@ -636,6 +639,28 @@ public boolean isIDPLoginFlowEnabled() {
...
+            for (ActivityInfo activityInfo : packageInfo.activities) {
+                if (activityInfo.name.equals(IDPAccountPickerActivity.class.getName())) {
```
Comment (bhariharan) [inline libs/SalesforceSDK/src/com/salesforce/androidsdk/app/SalesforceSDKManager.java]: Wouldn't this always be there anyway? Or does it depend on what's defined in the `Manifest` file?

File: libs/SalesforceSDK/src/com/salesforce/androidsdk/app/SalesforceSDKManager.java [hunk 3]
```diff
@@ -636,6 +639,28 @@ public boolean isIDPLoginFlowEnabled() {
         return !TextUtils.isEmpty(idpAppURIScheme);
     }
 
+    /**
+     * Returns whether this application is configured as a Identity Provider
+     * @return True - if
+     */
+
+    public boolean isIdentityProvider() {
```
Comment (bhariharan) [inline libs/SalesforceSDK/src/com/salesforce/androidsdk/app/SalesforceSDKManager.java]: Could we keep this API private?

**Comment by brandonpage** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">Tests have failed, see below for more information.</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Please provide a summary in the Pull Request description</td>
    </tr>
  </tbody>
</table>

### Tests: 

Name | Classname | Time|
--- | --- | ---|
testUpsertConcurrentEntries | com.salesforce.androidsdk.phonegap.SmartStoreLoadJSTest | 1.46|
testCleanResyncGhosts | com.salesforce.androidsdk.phonegap.SmartSyncJSTest | 352.823|
testSyncSObjectDetectConflictUpdate | com.salesforce.androidsdk.phonegap.SmartSyncJSTest | 0.353|
testSyncUpLocallyDeleted | com.salesforce.androidsdk.phonegap.SmartSyncJSTest | 0.327|
testSyncUpLocallyUpdatedWithNoOverwrite | com.salesforce.androidsdk.phonegap.SmartSyncJSTest | 0.377|
testDoesUserAccountExist | com.salesforce.androidsdk.accounts.UserAccountManagerTest | 0.152|
testLoadAccountMRUObjectsFromServer | com.salesforce.androidsdk.smartsync.manager.MetadataManagerTest | 5.344|
testLoadGlobalMRUObjectsFromServer | com.salesforce.androidsdk.smartsync.manager.MetadataManagerTest | 4.778|


<p align="right" data-meta="generated_by_ci/circleci">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

**Pull request merged**